clipper_admin/clipper_admin/clipper_admin.py

```diff
@@ -0,0 +1,995 @@
+        url = "http://{host}/admin/add_model".format(
+            host=self.cm.get_admin_addr())
+        if image is None:
+            image = CONTAINERLESS_MODEL_IMAGE
+        if labels is None:
+            labels = DEFAULT_LABEL
+        req_json = json.dumps({
+            "model_name": name,
+            "model_version": version,
+            "labels": labels,
+            "input_type": input_type,
+            "container_name": image,
+            "model_data_path": "DEPRECATED",
+        })
+        headers = {'Content-type': 'application/json'}
+        logger.debug(req_json)
+        r = requests.post(url, headers=headers, data=req_json)
+        if r.status_code != requests.codes.ok:
+            msg = "Received error status code: {code} and message: {msg}".format(
+                code=r.status_code, msg=r.text)
+            logger.error(msg)
+            raise ClipperException(msg)
+
+    def get_num_replicas(self, name, version):
+        """Gets the current number of model container replicas for a model.
+
+        Parameters
+        ----------
+        name : str
+            The name of the model
+        version : str, optional
+            The version of the model. If no version is provided,
+            the currently deployed version will be used.
+
+        Returns
+        -------
+        int
+            The number of active replicas
+
+        Raises
+        ------
+        :py:exc:`clipper.UnconnectedException`
+        """
+        if not self.connected:
+            raise UnconnectedException()
+        if version is None:
+            model_info = self.get_all_models(verbose=True)
+            for m in model_info:
+                if m["is_current_version"]:
+                    version = m["model_version"]
+        assert version is not None
+        version = str(version)
+        return self.cm.get_num_replicas(name, version)
+
+    def set_num_replicas(self, name, num_replicas, version=None):
+        """Sets the total number of active replicas for a model.
+
+        If there are more than the current number of replicas
+        currently allocated, this will remove replicas. If there are
+        less, this will add replicas.
+
+        Parameters
+        ----------
+        name : str
+            The name of the model
+        version : str, optional
+            The version of the model. If no version is provided,
+            the currently deployed version will be used.
+        num_replicas : int, optional
+            The desired number of replicas.
+
+        Raises
+        ------
+        :py:exc:`clipper.UnconnectedException`
+        :py:exc:`clipper.ClipperException`
+        """
+        if not self.connected:
+            raise UnconnectedException()
+        if version is None:
+            model_info = self.get_all_models(verbose=True)
+            for m in model_info:
+                if m["is_current_version"]:
+                    version = m["model_version"]
```

Comment:
This logic needs to be changed in the same way as lines 453-459. We should define a `get_current_model_version(name)` method to minimize duplicate code.

Reply:
Done